packages/myst-ext-icon/src/index.ts

```diff
@@ -0,0 +1,20 @@
+import type { RoleSpec, RoleData, GenericNode } from 'myst-common';
+
+export const iconRole: RoleSpec = {
+  name: 'fas',
+  alias: ['fab', 'far'],
```

Comment:
Would have to update the kind logic below, but I think the role makes more sense as an icon role with aliases?

```suggestion
  name: 'icon',
  alias: ['fab', 'fas', 'far'],
```

Reply:
@rowanc1 see #1361 for where I think this PR is going (I'm working locally and have changed tack slightly).